pilot/pkg/model/sidecar.go

```diff
@@ -33,11 +33,17 @@ const (
 	wildcardService   = host.Name("*")
 )
 
-var sidecarScopeKnownConfigTypes = map[resource.GroupVersionKind]struct{}{
-	gvk.ServiceEntry:    {},
-	gvk.VirtualService:  {},
-	gvk.DestinationRule: {},
-}
+var (
+	sidecarScopeKnownConfigTypes = map[resource.GroupVersionKind]struct{}{
+		gvk.ServiceEntry:    {},
+		gvk.VirtualService:  {},
+		gvk.DestinationRule: {},
+	}
+
+	sidecarScopeNamespaceConfigTypes = map[resource.GroupVersionKind]struct{}{
```

Comment:
So hypothetically, we can add EnvoyFilter and the authn/z stuff here and it will just work?

Reply:
I am not sure if all 3 security ones follow the same scoping, if they do I think this would cover all of our CRDs. Except gateway... we would need to do some custom logic there I think

Reply:
Do you think the scoping is broader or narrower? If narrower this will at least be an incremental improvement.

Reply:
I *think* all of them follow the exact same logic of root namespace -> namespace -> workload selector as sidecar. So in theory we could stick all 5 of them in and it would just work. We probably can just do sidecar in this PR though and then investigate more? Unless you like one big PR rather than smaller prs

Reply:
Assuming checks pass I think this one is good to go.